lineapy/execution/executor.py

```diff
@@ -239,6 +241,35 @@ def _execute(
             f"Don't know how to execute node type {type(node)}"
         )
 
+    @_execute.register
+    def _execute_if(
+        self,
+        node: IfNode,
+        changes: Iterable[TracebackChange],
+        variables: Optional[Dict[str, LineaID]],
+    ) -> PrivateExecuteResult:
+        return PrivateExecuteResult(
+            # An `execute if` uses the test condition to create a new control flow node  a global set by a call node so,
+            # Copy the result and the timing from the call node
+            self._id_to_value[node.test_id],
```

Comment:
i think test node is a boolop/compareop and will have its own privateexecuteresult that will be returned. execute_if should not return anything and should be similar to execute_else in that it should return none.